File: TP2/triangle/Triangle.java
package triangle;

public class Triangle {
    private double coteA;
    private double coteB;
    private double coteC;

    public int typeTriangle(){
        if (coteA > coteB+coteC || coteB > coteA+coteC || coteC > coteB+coteA || coteA <= 0 || coteB <= 0 || coteC <= 0)
            return -1;
        else if (coteA==coteB && coteA==coteC && coteB==coteC)
            return 3;
        else if (coteA==coteB || coteA==coteC || coteC==coteB)
            return 2;
        else return 1;
    }

    public double getCoteA() {
        return coteA;
    }

    public void setCoteA(double coteA) {
        this.coteA = coteA;
    }

    public double getCoteB() {
        return coteB;
    }

    public void setCoteB(double coteB) {
        this.coteB = coteB;
    }

    public double getCoteC() {
        return coteC;
    }

    public void setCoteC(double coteC) {
        this.coteC = coteC;
    }
}

File: TP1/triangle/triangle.c
/*
Zakari IKHOU 
Akim SAADI
Mohamed TAKHCHI
Reda CHANAA
Sohaib EL MOUHTADI 
12/01/2020
*/

#include <stdlib.h>
#include <stdio.h>
#include "triangle.h"

Triangle readData (char * file){
	FILE *f;
	f=fopen(file,"r");
	if (f == NULL)
	{
		
		Triangle triangle;
		triangle.coteA = -1;
		triangle.coteB = -1;
		triangle.coteC = -1;
		return triangle;
	} 
	int i;
	i=0;
	Triangle triangle;
	char ligne[50] = "";;
	while (fgets(ligne, 50, f) != NULL){
		switch(i)
	   {
		   case 0 : triangle.coteA = atoi(ligne); break;
		   case 1 : triangle.coteB = atoi(ligne); break;
		   case 2 : triangle.coteC = atoi(ligne); break;
		   default : ;
	   }
	   i++;
   }
   fclose(f);
   	if (i!=3)
	{
		Triangle triangle;
		triangle.coteA = -1;
		triangle.coteB = -1;
		triangle.coteC = -1;
		return triangle;
	} 
   return triangle;
}

int typeTriangle(float a, float b, float c){
	if (a > b+c || b > a+c || c > b+a || a <= 0 || b <= 0 || c <= 0)
		return -1;
	else if (a==b && a==c && b==c)
		return 3;
	else if (a==b || a==c || c==b)
		return 2;
	else return 1;
}
